src/unix/udp.c

```diff
@@ -257,23 +319,33 @@ static void uv__udp_recvmsg(uv_udp_t* handle) {
     h.msg_namelen = sizeof(peer);
     h.msg_iov = (void*) &buf;
     h.msg_iovlen = 1;
+#if defined(__linux__)
+    if (flag & MSG_ERRQUEUE) {
+      h.msg_control = control;
+      h.msg_controllen = sizeof(control);
+    }
+#endif
 
     do {
-      nread = recvmsg(handle->io_watcher.fd, &h, 0);
+      nread = recvmsg(handle->io_watcher.fd, &h, flag);
     }
     while (nread == -1 && errno == EINTR);
 
     if (nread == -1) {
       if (errno == EAGAIN || errno == EWOULDBLOCK)
         handle->recv_cb(handle, 0, &buf, NULL, 0);
-      else
-        handle->recv_cb(handle, UV__ERR(errno), &buf, NULL, 0);
```

Comment:
Removing this on non-linux platforms doesn't look Obviously Correct(TM). Is it?

Reply:
Yup, I'll bring this back :-)